src/common/events/events_io.cpp

```diff
@@ -137,6 +137,76 @@ mio::ExpectedSize ReaderFromAsyncReader::Read(
 	return read;
 }
 
+TeeReader::TeeReaderLeafPtr TeeReader::AttachSecondaryReader() {
+	auto reader = make_shared<TeeReaderLeaf>(shared_from_this());
+	leaf_readers_.insert({reader, TeeReaderRequestedRead {}});
+	return reader;
+}
+
+error::Error TeeReader::ReadyToAsyncRead(
+	TeeReader::TeeReaderLeafPtr leaf_reader,
+	vector<uint8_t>::iterator start,
+	vector<uint8_t>::iterator end,
+	mio::AsyncIoHandler handler) {
+	// TODO: should this be an assert() instead?
```

Comment:
I suggest [`AssertOrReturnError`](https://github.com/mendersoftware/mender/blob/master/src/common/error.hpp#L24).

Reply:
Thank you